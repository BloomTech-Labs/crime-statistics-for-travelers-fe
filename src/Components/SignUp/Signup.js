import React from "react";
import {Link} from 'react-router-dom'
import { withFormik, Form, Field } from "formik";
import * as Yup from "yup";
import axios from "axios";
import { Redirect } from "react-router-dom";
import Swal from "sweetalert2";
import styled from "styled-components";
import "../../App.css"
// import posed from "react-pose";

const H5 = styled.h5`
  color: #b22222;
`;

const FormContainer = styled.div`
  

`;
const BTN =styled.button`
 
`
const Btn = styled(Link)``

// const Label = styled.label`
//   color: #f3e367;
//   font-size: 2.5em;
// `;

function Signup({ touched, errors }) {
 



  return (
    <Form className="form">
      <FormContainer>

        <Field
          id="name"
          type="name"
          autoComplete="off"
          placeholder="name"
          name="name"
        />
        <H5>{touched.name && errors.name}</H5>
  
        <Field
          id="email"
          type="email"
          autoComplete="off"
          placeholder="email"
          name="email"
        />
        <H5>{touched.email && errors.email}</H5>

        <Field
          id="password"
          type="password"
          autoComplete="off"
          placeholder="password"
          name="password"
        />
        <H5>{touched.password && errors.password}</H5> 
        <BTN  className="formBTN" type="submit">
          <Btn to='/Login'>Signup</Btn>
        </BTN>
      </FormContainer>
    </Form>
  );
}

export default withFormik({
  mapPropsToValues() {
    return {
      name: "",
      email:"",
      password: ""
      // confirmPassword: "",
      // rememberPassword: false
    };
  },
  validationSchema: Yup.object().shape({
    username: Yup.string()
      .min(3, "Must be 3 characters or more")
      .max(20, "Must be less than 20 characters")
      .required("This field is required"),
      email: Yup.string()
      .required("please enter an email to continue"),
    password: Yup.string()
      .min(2, "Must be 3 characters or more")
      .max(100, "Must be less than 100 characters")
      .required("Enter a password to continue")
    // passwordConfirmation: Yup.string()
    // .oneOf([Yup.ref('password'), null], 'Passwords must match'),
  }),
  handleSubmit(values, formikBag) {
    console.log(values,"Login values")
    axios
      .post(`https://usemytechstuff.herokuapp.com/api/auth/signUp/`, values)
      .then((response) => {
        localStorage.setItem('token', response.data.payload);
        console.log('does token data exist:', response.data.payload)
        formikBag.props.history.push('/');
        formikBag.props.setToken(response.data.payload)
      })
      .catch((e) => {
        // console.log(e.response.data && response.data);
      });
    //   {if(token === null){
    //     Swal.fire({
    //         position: 'center',
    //         type: 'error',
    //         title: 'Try Again!',
    //         showConfirmButton: false,
    //         timer: 2500
    //       })
    //   }else{
        Swal.fire({
            position: 'center',
            type: 'success',
            title: 'Registering User...',
            showConfirmButton: false,
            timer: 2500
          })
    //   }}

  }

})(Signup);