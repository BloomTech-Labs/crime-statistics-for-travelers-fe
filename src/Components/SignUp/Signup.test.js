import React from "react";
import Signup from "./Signup";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

it("Snapshot test for Signup page", () => {
  const tree = renderer
    .create(
      <Router>
        <Signup />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
