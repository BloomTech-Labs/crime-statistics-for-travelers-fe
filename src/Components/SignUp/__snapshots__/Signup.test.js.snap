// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Signup page 1`] = `
<div
  className="col-container"
>
  <div
    className="col1"
  >
    <div
      className="heading-container"
    >
      <h1
        className="css-1kdk8in"
      >
        Where to next?
      </h1>
      <strong>
        <p>
          Welcome Back!
        </p>
        <p>
          Please sign in 
        </p>
        <p>
          with your info.
        </p>
      </strong>
      <a
        href="/login"
        onClick={[Function]}
      >
        <button
          className="css-3n2czx"
        >
          Sign In
        </button>
      </a>
    </div>
  </div>
  <div
    className="col2"
  >
    <div
      className="form-container css-192m1qh"
    >
      <h1
        className="heading"
      >
        Create an account.
      </h1>
      <div
        className="auth-links"
      >
        <div
          className="auth-link"
        >
          <img
            alt="google auth0 button"
            src="btn_google_signin_light_normal_web.png"
          />
        </div>
        <div
          className="auth-link"
        >
          <img
            alt="facebook auth0 button"
            src="login-facebook.png"
          />
        </div>
      </div>
      <form
        className="signup-form"
        onSubmit={[Function]}
      >
        <input
          className="form-input"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          value=""
        />
        <input
          className="form-input"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          value=""
        />
        <input
          className="form-input"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <button
          className="css-17ddmg5"
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</div>
`;
